src/states/node.rs

```diff
@@ -2260,12 +2264,48 @@ impl Node {
         Ok(())
     }
 
+    // If multiple nodes dropped from our section, duplicated `OwnSectionMerge` will be sent,
+    // containing different dropped out nodes. This will result in differnt `OtherSectionMerge` to
+    // be sent out, hence causing non-accumulation on the receiver side.
+    // So here we will take a union of our_merged_section from all the `OwnSectionMerge`, and resend
+    // `OtherSectionMerge` whenever the new union is different to the last one sent out.
+    fn union_our_merged_section(&mut self, merge_prefix: Prefix<XorName>, sections: &SectionMap) {
+        let mut our_merged_section = BTreeSet::new();
+        for (ver_pfx, peers) in sections.iter() {
+            if ver_pfx.prefix().is_extension_of(&merge_prefix) {
+                our_merged_section.extend(peers.into_iter().map(|peer| *peer.name()));
+            }
+        }
+        match self.our_merged_section_cache.remove(&merge_prefix) {
+            Some((send_other_section_merge, prev_our_merged_section)) => {
+                our_merged_section.extend(prev_our_merged_section.iter());
```

Comment:
Ah, right. Then you could just compare `our_merged_section.len()` before and after the `extend` call to avoid the cloning. But maybe that's premature optimisation…